cli/Valet/Site.php

```diff
@@ -185,13 +185,68 @@ function secure($url)
 
         $this->files->ensureDirExists($this->certificatesPath(), user());
 
+        $this->createCa();
+
         $this->createCertificate($url);
 
         $this->files->putAsUser(
             VALET_HOME_PATH.'/Nginx/'.$url, $this->buildSecureNginxServer($url)
         );
     }
 
+    /**
+     * If CA and root certificates are nonexistent, crete them and trust the root cert.
+     *
+     * When created CN and O fields receive random affixes. The random part is saved in file so that later certificate can be untrusted.
+     *
+     * @return void
+     */
+    function createCa()
+    {
+        $caPemPath = $this->certificatesPath().'/LaravelValetCASelfSigned.pem';
+        $caKeyPath = $this->certificatesPath().'/LaravelValetCASelfSigned.key';
+        $caAffixPath = $this->certificatesPath().'/LaravelValetCASelfSigned.affix';
+
+        if ($this->files->exists($caKeyPath) && $this->files->exists($caPemPath) && $this->files->exists($caAffixPath)) {
+            return;
+        }
+
+        $oName = 'Laravel Valet CA Self Signed O';
+        $cName = 'Laravel Valet CA Self Signed C';
+        $affix = '';
+
+        if ($this->files->exists($caKeyPath)) {
+            $this->files->unlink($caKeyPath);
+        }
+        if ($this->files->exists($caPemPath)) {
+            $this->files->unlink($caPemPath);
+        }
+        if ($this->files->exists($caAffixPath)) {
+            $affix = $this->files->get($caAffixPath);
+            $this->cli->run(sprintf(
+                'sudo security delete-certificate -c "%s%s" /Library/Keychains/System.keychain -t',
+                $cName, $affix
+            ));
+            $this->files->unlink($caAffixPath);
+        }
+        $this->cli->run(sprintf(
+            'sudo security delete-certificate -c "%s" /Library/Keychains/System.keychain -t',
+            $cName
+        ));
+
+        $affix = bin2hex(openssl_random_pseudo_bytes(15));
+        $this->files->putAsUser($caAffixPath, $affix);
+
+        $oName .= $affix;
+        $cName .= $affix;
+
+        $this->cli->runAsUser(sprintf(
+            'openssl req -new -newkey rsa:2048 -days 730 -nodes -x509 -subj "/C=LV/ST=Latvia/O=%s/localityName=Riga/commonName=%s/organizationalUnitName=Developers/emailAddress=noreply@valet.test/" -keyout %s -out %s',
```

Comment:
Can this be done with less geographic specificity?

Reply:
It can be changed. Should it? What do you suggest - some random state (again) in US is a better choice, because...? Personally I see no harm when coder references to himself, unless it does not bite functionality (here it does not). It is my easter egg (come visit), however people with write access in this repo still will do as they see fit.

Reply:
Was just asking a question.

Happy to celebrate the easter egg. :)

Reply:
Is there some reason the (basically empty) `subj` that is [currently being used for signing requests](https://github.com/laravel/valet/blob/4c243d7d71a0c3fa9cb850d6f90d40ac11bfde91/cli/Valet/Site.php#L239-L242) can’t be used here as well?

```php
        $this->cli->runAsUser(sprintf(
            'openssl req -new -key %s -out %s -subj "/C=/ST=/O=/localityName=/commonName=*.%s/organizationalUnitName=/emailAddress=/" -config %s -passin pass:',
            $keyPath, $csrPath, $url, $confPath
        ));
```

Reply:
@Zegnat none that I know of; C,ST,L can be left empty. Considering this line of code raises so many [questions](http://s2.quickmeme.com/img/43/43c6005bf1e35d4742ea9a46341aca9c2883871c37205f84927f95d8d849a907.jpg) I just pushed update where C, ST, L are empty, no time should be spent on this any more.